From 048f0a9a2c5aa89ac7db3a9ddfc86ed85a8b26df Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Thu, 17 Apr 2025 12:51:45 +0900
Subject: [PATCH 7/9] HACK: ATF missing parts support

This patch make to avoid following error:
`"No ARM Trusted Firmware AArch64 Firmware Image"`
And, adds some missing parts for ATF.

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 boot/image-fit.c     | 1 +
 common/spl/spl_fit.c | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/boot/image-fit.c b/boot/image-fit.c
index 41ab1f552b0..7a9fdb9536b 100644
--- a/boot/image-fit.c
+++ b/boot/image-fit.c
@@ -2183,6 +2183,7 @@ int fit_image_load(struct bootm_headers *images, ulong addr,
 		image_type == IH_TYPE_FPGA ||
 		fit_image_check_os(fit, noffset, IH_OS_LINUX) ||
 		fit_image_check_os(fit, noffset, IH_OS_U_BOOT) ||
+		fit_image_check_os(fit, noffset, IH_OS_ARM_TRUSTED_FIRMWARE) ||
 		fit_image_check_os(fit, noffset, IH_OS_TEE) ||
 		fit_image_check_os(fit, noffset, IH_OS_OPENRTOS) ||
 		fit_image_check_os(fit, noffset, IH_OS_EFI) ||
diff --git a/common/spl/spl_fit.c b/common/spl/spl_fit.c
index 49b4df60560..b7be4645b83 100644
--- a/common/spl/spl_fit.c
+++ b/common/spl/spl_fit.c
@@ -372,6 +372,7 @@ static bool os_takes_devicetree(uint8_t os)
 {
 	switch (os) {
 	case IH_OS_U_BOOT:
+	case IH_OS_ARM_TRUSTED_FIRMWARE:
 		return true;
 	case IH_OS_LINUX:
 		return IS_ENABLED(CONFIG_SPL_OS_BOOT) ||
@@ -560,6 +561,8 @@ static int spl_fit_image_get_os(const void *fit, int noffset, uint8_t *os)
 	 */
 	if (!strcmp(name, "u-boot"))
 		*os = IH_OS_U_BOOT;
+        else if (!strcmp(name, "arm-trusted-firmware"))
+		*os = IH_OS_ARM_TRUSTED_FIRMWARE ;
 	else
 		*os = IH_OS_INVALID;
 
-- 
2.34.1

